### DISCUSSION TITLE
Metadata for known Special Folders

### DISCUSSION THREAD

**Discussion by michaeljfarr** [personal-vault/pvp#13] [Q&A]
Do you currently possess or are you planning to develop a method for categorizing and classifying folders and files based on information derived from the surrounding folder hierarchy?

For instance, consider the scenario where a pattern like "/folder/.git" is identified. This observation would lead to the inference that all elements within the "folder" are part of a Git repository.

Git's pattern-based approach is also employed by various other folder-oriented projects (such as IntelliJ IDEA and others). However, numerous alternative patterns exist, such as distinctive parent folders (e.g., DCIM, node_modules), layouts adhering to Windows and FHS standards, connections to the Windows registry (like Google Drive folders), and associations with fstab entries (like mount points).

While achieving a flawless solution for discerning the purpose of each folder might not be feasible, the implementation of a database or an API to manage such a database would prove advantageous. Presently, certain online resources cater to file-related information (such as https://www.file-extensions.org/ and https://fileinfo.com/), yet comparable systems with a focus on directories appear to be lacking.

**Comment by dlucian** [comment]
Yes, that's exactly what I stumbled upon while creating a mental model of this system. In my case, I was thinking about folders where I created the content (for example my notes, my documents, the photos that I took) and content that I'm just storing (for example an archive of something created by others). 

So I got to a point where using special dot-files I would add context information in `.context.json` files which add meta-data to the contents of that folder (and all sub-folders), such as `{"author":"John Doe","description":"Greece vacation photos"}` so that when indexing the contents of that folder, we have additional meta information which will turn out to be useful when looking for data.  

What I'm avoiding now is defining the whole structure of these context files until I see how indexing and retrieving data works, as the sole purpose of adding meta-information to folders is to make the information easily retrievable in a relevant and valuable manner.
